modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/tabletoolbar/plugin.js

```diff
@@ -0,0 +1,1475 @@
+					exec(editor) {
+						placeCursorInCell(mergeCells(editor.getSelection(), "right"), true);
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellMergeDown",
+				},
+				createDef({
+					allowedContent: "td[rowspan]",
+					requiredContent: "td[rowspan]",
+					exec(editor) {
+						placeCursorInCell(mergeCells(editor.getSelection(), "down"), true);
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellVerticalSplit",
+				},
+				createDef({
+					allowedContent: "td[rowspan]",
+					requiredContent: "td[rowspan]",
+					exec(editor) {
+						placeCursorInCell(verticalSplitCell(editor.getSelection()));
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellHorizontalSplit",
+				},
+				createDef({
+					allowedContent: "td[colspan]",
+					requiredContent: "td[colspan]",
+					exec(editor) {
+						placeCursorInCell(horizontalSplitCell(editor.getSelection()));
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellInsertBefore",
+				},
+				createDef({
+					requiredContent: "table",
+					exec(editor) {
+						const selection = editor.getSelection();
+						insertCell(selection, true);
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellInsertAfter",
+				},
+				createDef({
+					requiredContent: "table",
+					exec(editor) {
+						const selection = editor.getSelection();
+						insertCell(selection);
+					},
+				})
+			);
+		},
+		getSelectedCells,
+	});
+
+	ListBox = CKEDITOR.tools.createClass({
+		$: function (editor, config, clickFn) {
+			this.$ = new CKEDITOR.dom.element("div");
+
+			this.clickFn = clickFn;
+
+			this.setHtml(config);
+		},
+		proto: {
+			getElement: function () {
+				return this.$;
+			},
+
+			getHtml: function () {
+				return this.getElement().getOuterHtml();
+			},
+
+			setHtml: function (config) {
+				this.getElement().setHtml(
+					"<ul class='listbox' id='" +
+						config.panelId +
+						"' role='listbox'>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[0].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[0].commandName +
+						"</button>" +
+						"</li>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[1].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[1].commandName +
+						"</button>" +
+						"</li>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[2].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[2].commandName +
+						"</button>" +
+						"</li>" +
+						"</ul>"
+				);
+			},
+		},
+	});
+})();
```

Comment:
Do we need `ListBox` class? It looks to me like over-engineering. It is used only locally, in one place, `renderPanel`. Instead, we can just build HTML inside `renderPanel`.

Reply:
when using CKEditor's templating system I need to create another plugin, isn't? ~~So, I'll need to create a new plugin called `uipanel`.~~

I think not, It's just apply `config` property as parameters and use template.output as a output of renderPanel function. :)

Reply:
How can I iterate, like using a map inside a template?

We don't have an example for it in [template.js tests](https://github.com/ckeditor/ckeditor4/blob/cae20318d46745cc46c811da4e7d68b38ca32449/tests/core/template.js#L108)

Reply:
Done ✅

Reply:
f18203c